pkg/controllermanager/controller/managedseedset/managedseedset_replicagetter.go

```diff
@@ -71,17 +69,13 @@ func (rg *replicaGetter) GetReplicas(ctx context.Context, set *seedmanagementv1a
 		return nil, err
 	}
 
-	// If CachedRuntimeClients feature gate is enabled, cross-check number of shoots with a partial metadata list
-	// from the API server to ensure what we got from the cache is up-to-date
-	if controllermanagerfeatures.FeatureGate.Enabled(features.CachedRuntimeClients) {
```

Comment:
You can remove this line now:
https://github.com/gardener/gardener/blob/31b8385f6fbae62af46e18f9c55a6e7624cec16e/pkg/controllermanager/controller/managedseedset/managedseedset_replicagetter_test.go#L126